pkg/apis/core/validation/cloudprofile.go

```diff
@@ -44,11 +45,68 @@ func ValidateCloudProfileUpdate(newProfile, oldProfile *core.CloudProfile) field
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newProfile.ObjectMeta, &oldProfile.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateCloudProfileSpecUpdate(&newProfile.Spec, &oldProfile.Spec, field.NewPath("spec"))...)
 	allErrs = append(allErrs, ValidateCloudProfile(newProfile)...)
 
 	return allErrs
 }
 
+// ValidateCloudProfileSpecUpdate validates the spec update of a CloudProfile
+func ValidateCloudProfileSpecUpdate(new *core.CloudProfileSpec, old *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateCloudProfileVersionsUpdate(new.Kubernetes.Versions, old.Kubernetes.Versions, fldPath.Child("kubernetes", "versions"))...)
+
+	for _, oldImage := range old.MachineImages {
+		for index, newImage := range new.MachineImages {
+			if oldImage.Name == newImage.Name {
+				allErrs = append(allErrs, validateCloudProfileVersionsUpdate(newImage.Versions, oldImage.Versions, fldPath.Child("machineImages").Index(index).Child("versions"))...)
+			}
+		}
+	}
+
+	return allErrs
+}
+
+// ValidateCloudProfileAddedVersions validates versions added to the CloudProfile
+func ValidateCloudProfileAddedVersions(versions []core.ExpirableVersion, addedVersions map[int]core.ExpirableVersion, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for index, version := range addedVersions {
+		// Do not allow adding addedVersions with expiration date in the past
+		if  version.ExpirationDate != nil && version.ExpirationDate.Time.UTC().Before(time.Now().UTC()) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Index(index), fmt.Sprintf("unable to add version %q. Adding a version with expiration date in the past is not allowed", version.Version)))
+			continue
+		}
+
+		if version.Classification != nil && *version.Classification == core.ClassificationSupported {
+			currentSemVer, err := semver.NewVersion(version.Version)
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(index), version.Version, fmt.Sprintf("unable to add version '%s'. Version must be a semantic version", version.Version)))
+				continue
+			}
+
+			filteredVersions, err := helper.FindVersionsWithSameMajorMinor(helper.FilterVersionsWithClassification(versions, core.ClassificationSupported), *currentSemVer, )
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(index), currentSemVer.String(), fmt.Sprintf("unable to add version '%s' : '%v'", version.Version, err)))
```

Comment:
> I wonder if we can ignore this error since the semver check is done by the validator

Not returning an error any more according to this suggestion - so this error is no longer needed.